websites/Y/YukiList/presence.ts

```diff
@@ -0,0 +1,164 @@
+      }
+
+      if (showButtons && !privacyMode) {
+        presenceData.buttons = [
+          {
+            label: 'View Anime on YukiList',
+            url: document.location.href,
+          },
+        ]
+      }
+    }
+    else {
+      presenceData.details = `${strings.viewing} Anime`
+      presenceData.state = 'Loading...'
+    }
+  }
+  else if (pathname === '/search') {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = `${strings.searching} for Anime`
+    presenceData.smallImageKey = Assets.Search
+    presenceData.smallImageText = strings.searching
+  }
+  else if (pathname === '/upcoming') {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = 'Viewing Upcoming Anime'
+  }
+  else if (pathname === '/completed') {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = 'Viewing Completed Anime'
+  }
+  else if (pathname.startsWith('/me/profile')) {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = privacyMode ? 'Viewing Profile' : 'Viewing My Profile'
+  }
+  else if (pathname.startsWith('/user/')) {
+    const username = document.querySelector('.text-3xl.font-bold.text-white, h1.text-3xl')?.textContent?.trim()
+    presenceData.details = 'Viewing Profile'
+    presenceData.state = privacyMode || !username ? 'Someone\'s profile' : username
+
+    if (!privacyMode) {
+      const profilePic = document.querySelector<HTMLImageElement>('.w-24.h-24.rounded-2xl img, .rounded-2xl img[alt*="User"]')
+      if (profilePic?.src && !profilePic.src.includes('data:')) {
+        presenceData.largeImageKey = profilePic.src
+        presenceData.smallImageKey = 'https://i.imgur.com/MdbUxWe.png'
+        presenceData.smallImageText = 'YukiList'
+      }
+    }
+  }
+  else if (pathname.startsWith('/me/list')) {
+    const searchParams = new URLSearchParams(window.location.search)
+    const status = searchParams.get('status')
+    const statusMap: { [key: string]: string } = {
+      watching: 'Watching',
+      completed: 'Completed',
+      on_hold: 'On Hold',
+      dropped: 'Dropped',
+      plan_to_watch: 'Plan to Watch',
+    }
+
+    const countText = document.querySelector('.text-sm.text-muted-foreground')?.textContent
+    const countMatch = countText?.match(/(\d+)\s+anime/)
+    const count = countMatch ? countMatch[1] : null
+
+    const statusLabel = status && statusMap[status] ? statusMap[status] : 'Watching'
+
+    presenceData.details = 'Managing Watchlist'
+    presenceData.state = count ? `${statusLabel} (${count})` : `${statusLabel} List`
+  }
+  else if (pathname.startsWith('/auth')) {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = 'Signing In'
+  }
+  else {
+    presenceData.details = `${strings.browsing} YukiList`
+    presenceData.state = 'Exploring'
+  }
+
+  if (presenceData.details)
+    presence.setActivity(presenceData)
+  else presence.setActivity()
```

Comment:
The current code would show a blank activity with the activity's name and image, rather than clearing the activity. If you mean to clear the contents, do this
```suggestion
  else presence.clearActivity()
```